src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js

```diff
@@ -154,44 +191,22 @@ const ContentCreateWidget = () => {
                         </div>
                         <div className="ibexa-instant-filter__desc">{filtersDescLabel}</div>
                         <div className="ibexa-instant-filter__items">
-                            {contentTypesWithSuggestions.map(([groupName, groupItems], index) => {
+                            {contentTypesDataToShow.map(([groupName, groupItems], index) => {
                                 const isSuggestionGroup = !!suggestions.length && index === 0;
-                                const restrictedContentTypeIds = selectedLocation?.permissions?.create.restrictedContentTypeIds ?? [];
-                                const isHiddenGroup = groupItems.every((groupItem) => {
-                                    const isNotSearchedName = filterQuery && !groupItem.name.toLowerCase().includes(filterQuery);
-                                    const hasNotPermission =
-                                        restrictedContentTypeIds.length && !restrictedContentTypeIds.includes(groupItem.id.toString());
-                                    const isNotAllowedContentType =
-                                        allowedContentTypes && !allowedContentTypes.includes(groupItem.identifier);
-                                    const isHiddenByConfig = groupItem.isHidden;
-
-                                    return (
-                                        isNotSearchedName ||
-                                        hasNotPermission ||
-                                        isNotAllowedContentType ||
-                                        isHiddenByConfig ||
-                                        (isNotSearchedName && isSuggestionGroup)
-                                    );
+                                const visibleGroupItems = groupItems.filter((groupItem) => {
+                                    const isSearchedName = !filterQuery || groupItem.name.toLowerCase().includes(filterQuery);
```

Comment:
little possible optimization - put `filterQuery` outside of filter, as if it's empty, there's no need it do any filtering.
But that's just nitpick :)